execution/execution-correctness/src/local.rs

```diff
@@ -11,37 +13,65 @@ use std::{
     sync::{Arc, Mutex},
 };
 
+pub struct LocalService {
+    block_executor: Box<dyn BlockExecutor>,
+    prikey: Option<Ed25519PrivateKey>,
+}
+
+impl LocalService {
+    pub fn new(block_executor: Box<dyn BlockExecutor>, prikey: Option<Ed25519PrivateKey>) -> Self {
+        Self {
+            block_executor,
+            prikey,
+        }
+    }
+}
+
 /// A local interface into ExecutionCorrectness. Constructed in such a way that the container / caller
 /// cannot distinguish this API from an actual client/server process without being exposed to
 /// the actual container instead the caller can access a Box<dyn TExecutionCorrectness>.
 pub struct LocalClient {
-    internal: Arc<Mutex<Box<dyn BlockExecutor>>>,
+    internal: Arc<Mutex<LocalService>>,
 }
 
 impl LocalClient {
-    pub fn new(internal: Arc<Mutex<Box<dyn BlockExecutor>>>) -> Self {
+    pub fn new(internal: Arc<Mutex<LocalService>>) -> Self {
         Self { internal }
     }
 }
 
-impl BlockExecutor for LocalClient {
+impl ExecutionCorrectness for LocalClient {
     fn committed_block_id(&mut self) -> Result<HashValue, Error> {
-        self.internal.lock().unwrap().committed_block_id()
+        self.internal
+            .lock()
+            .unwrap()
+            .block_executor
+            .committed_block_id()
     }
 
     fn reset(&mut self) -> Result<(), Error> {
-        self.internal.lock().unwrap().reset()
+        self.internal.lock().unwrap().block_executor.reset()
     }
 
     fn execute_block(
         &mut self,
-        block: (HashValue, Vec<Transaction>),
+        block: Block,
         parent_block_id: HashValue,
     ) -> Result<StateComputeResult, Error> {
-        self.internal
-            .lock()
-            .unwrap()
-            .execute_block(block, parent_block_id)
+        let mut local = self.internal.lock().unwrap();
+        let mut result = local
+            .block_executor
+            .execute_block(id_and_transactions_from_block(&block), parent_block_id)?;
+        if let Some(prikey) = local.prikey.as_ref() {
+            let vote_proposal = VoteProposal::new(
+                result.extension_proof(),
+                block,
+                result.epoch_state().clone(),
+            );
+            let signature = prikey.sign_message(&vote_proposal.hash());
+            result.set_signature(signature);
+        }
```

Comment:
so why the copy pasta? why not have a common interface like we have in safety rules?

Reply:
What do you mean here? Do you mean why we have `ExecutionCorrectness` trait and `BlockExecutor` trait, both of which look so similar. The reason is the signature of `execute_block` is different. the second paramater is `Block` and `Vec<Transaction>`, respectively.

Reply:
ugh, I lost context on this comment and don't see why it's here....